src/problems/CurrentSheet/test_current_sheet.cpp

```diff
@@ -0,0 +1,132 @@
+	static constexpr bool is_hydro_enabled = true;
+	static constexpr int numMassScalars = 0;		     // number of mass scalars
+	static constexpr int numPassiveScalars = numMassScalars + 0; // number of passive scalars
+	static constexpr bool is_radiation_enabled = false;
+	static constexpr bool is_mhd_enabled = true;
+	static constexpr int nGroups = 1; // number of radiation groups
+	static constexpr UnitSystem unit_system = UnitSystem::CGS;
+};
+
+// constants
+constexpr double gamma_gas = quokka::EOS_Traits<CurrentSheet>::gamma;
+constexpr double beta = 0.1;
+constexpr double A = 0.1;
+
+// background states
+constexpr double rho0 = 1.0;
+constexpr double P0 = 0.5 * beta;
+
+template <> void QuokkaSimulation<CurrentSheet>::setInitialConditionsOnGrid(quokka::grid const &grid_elem)
+{
+	// extract grid information
+	const amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> dx = grid_elem.dx_;
+	const amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> prob_lo = grid_elem.prob_lo_;
+	const amrex::Array4<double> &state_cc = grid_elem.array_;
+	const amrex::Box &indexRange = grid_elem.indexRange_;
+
+	amrex::ParallelFor(indexRange, [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+		const double y = prob_lo[1] + (i + 0.5) * dx[1]; // NOLINT(readability-math-missing-parentheses)
+
+		const double vx = A * std::sin(2.0 * M_PI * y);
+		const double vy = 0;
+
+		const double Ekin = 0.5 * rho0 * (vx*vx + vy*vy);
+		const double Eint = P0 / (gamma_gas - 1.0);
+		const double Emag = 0.5;
+
+		state_cc(i, j, k, HydroSystem<CurrentSheet>::density_index) = rho0;
+		state_cc(i, j, k, HydroSystem<CurrentSheet>::x1Momentum_index) = rho0 * vx;
+		state_cc(i, j, k, HydroSystem<CurrentSheet>::x2Momentum_index) = rho0 * vy;
+		state_cc(i, j, k, HydroSystem<CurrentSheet>::x3Momentum_index) = 0;
+		state_cc(i, j, k, HydroSystem<CurrentSheet>::internalEnergy_index) = Eint;
+		state_cc(i, j, k, HydroSystem<CurrentSheet>::energy_index) = Eint + Ekin + Emag;
+	});
+}
+
+template <> void QuokkaSimulation<CurrentSheet>::setInitialConditionsOnGridFaceVars(quokka::grid const &grid_elem)
+{
+	const amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> dx = grid_elem.dx_;
+	const amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> prob_lo = grid_elem.prob_lo_;
+	const amrex::Array4<double> &state_fc = grid_elem.array_;
+	const amrex::Box &indexRange = grid_elem.indexRange_;
+	const quokka::direction dir = grid_elem.dir_;
+
+	amrex::ParallelFor(indexRange, [=] AMREX_GPU_DEVICE(int i, int j, int k) {
+		const double x = prob_lo[0] + (i + 0.5) * dx[0]; // NOLINT(readability-math-missing-parentheses)
+
+		double by = NAN;
+		if (std::abs(x) > 0.25) {
+			by = 1.;
+		} else {
+			by = -1.;
+		}
+
+		// somehow this does not work -- the outputs show that x-BField is nonzero...
+		if (dir == quokka::direction::x) {
+			state_fc(i, j, k, Physics_Indices<CurrentSheet>::mhdFirstIndex) = 0;
+		} else if (dir == quokka::direction::y) {
+			state_fc(i, j, k, Physics_Indices<CurrentSheet>::mhdFirstIndex) = by;
```

Comment:
@AstroKriel There is an indexing bug somewhere. When I look at the plotfile output for the initial conditions, this component ends up in `x-BField` instead of `y-BField`. Can you take a look at what's going on?

Reply:
Switching the x and y components didn't work. When I tried that, both Bx and By were zero in the plotfile.

Reply:
Or am I making an obvious mistake that I'm not seeing?

Reply:
I wonder if there is just a bug in how grid_elem.dir_ is being set -- that would account for the symptoms. But it is also possible there is still a bug in plotfile writing. We found and fixed a similar bug earlier in the process, where the MF components were being matched to the wrong string in the output, so it could be a similar issue here.

Reply:
I've verified that the arrays are being set correctly with printf debugging. Where was the bug in the plotfile writing before?

Reply:
Ok, it looks like that bug is still present. The cell-centered output multifab is filled in order of directions {x,y,z}, then in order of components {velocity, B}, whereas the output labels are constructed in the opposite order. So the components of *-RiemannSolverVelocity and *-BField are mixed up.

Reply:
This is how it's saved in `AMRSimulation::PlotFileMFAtLevel`:
```
	if constexpr (Physics_Indices<problem_t>::nvarTotal_fc > 0) {
		for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
			AverageFCToCC(plotMF, state_new_fc_[lev][idim], idim, comp, 0, ncomp_per_dim_fc);
			comp += ncomp_per_dim_fc;
		}
	}
```
whereas this is how the labels are ordered:
```
	// add face-centered velocities
	for (int idim = 0; idim < AMREX_SPACEDIM; idim++) {
		componentNames_fc_.push_back({quokka::face_dir_str[idim] + "-RiemannSolverVelocity"});
	}
	// add mhd state variables
	if constexpr (Physics_Traits<problem_t>::is_mhd_enabled) {
		for (int idim = 0; idim < AMREX_SPACEDIM; idim++) {
			componentNames_fc_.push_back({quokka::face_dir_str[idim] + "-BField"});
		}
	}
```

Reply:
Yup, that is exactly the bug we fixed. So maybe somehow @AstroKriel didn't check it in? In any event, glad that was easy to fix.